kaska/s2_observations.py

```diff
@@ -111,75 +133,95 @@ def define_output(self):
-        # new_geoT[0] = new_geoT[0] + self.ulx*new_geoT[1]
-        # new_geoT[3] = new_geoT[3] + self.uly*new_geoT[5]
-        return proj, geoT.tolist(), nx, ny  # new_geoT.tolist()
-
-    def _find_granules(self, parent_folder, time_grid=None):
-        """Finds granules. Currently does so by checking for
-        Feng's AOT file."""
-        # this is needed to follow symlinks
-        test_files = [
-            x for f in parent_folder.iterdir() for x in f.rglob("**/*_aot.tif")
-        ]
-        try:
-            dates = [
-                    dt.datetime(*(list(map(int, f.parts[-5:-2]))))
-                    for f in test_files
-                ]
-        except ValueError:
-            dates = [
-                dt.datetime.strptime(
-                    f.parts[-1].split("_")[1], "%Y%m%dT%H%M%S"
-                )
-                for f in test_files
-            ]
-        # Sort dates by time, as currently S2A/S2B will be part of ordering
-        
-        #test_files = sorted(test_files, key=lambda x:dates[test_files.index(x)])
-        #dates = sorted(dates)
-        if time_grid is not None:
-            start_date = time_grid[0]
-            end_date = time_grid[-1]
-            self.dates = [d.replace(hour=0, minute=0, second=0) for d in dates
-                            if (d >= start_date) and (d <= end_date)] 
-            test_files = [test_files[i] for i, d in enumerate(dates)
-                            if (d >= start_date) and (d <= end_date)]
-        else:
-            self.dates = [x.replace(hour=0,minute=0, second=0) for x in dates]
-        temp_dict = dict(zip(self.dates, [f.parent for f in test_files]))
-        dates = sorted(self.dates)
-        self.date_data = {k:temp_dict[k] for k in dates}
-        self.dates = dates
-        
-        #self.date_data = dict(zip(self.dates, [f.parent for f in test_files]))
-        self.bands_per_observation = {}
-        LOG.info(f"Found {len(test_files):d} S2 granules")
+
+        return proj, geoT.tolist(), nx, ny
+
+    def _find_granules(self, time_grid=None):
+        """Finds granules within the given time grid if given.
+
+        Parameters
+        ----------
+        time_grid : A list of dates (optional).If not given, all the timestamps
+                    will be checked.
+        Returns
+        -------
+        None
+        Doesn't return anything, but changes `self.date_data`
+        """
+        folders = sorted([x for f in self.parent.iterdir()
+                          for x in f.rglob("*.SAFE")
+                          if x.name.find("MSIL1C") >= 0])
+
+        # Apply multithreaded
+        date_files = []
+        with ThreadPoolExecutor(max_workers=10) as executor:
+            for params in executor.map(
+                    lambda x: self._process_xml(x/"MTD_MSIL1C.xml"), folders):
+                date_files.append(params)
+
+        dates = [x[0].replace(hour=0, minute=0, second=0, microsecond=0)
+                 for x in date_files]
```

Comment:
I don't think we need the time, but @jgomezdans will know for sure.

Reply:
We don't need the time per se, but there's one function used later that works only for `datetime`, and not `date`, that's why all timesteps are `datetime`s.
And the time has to be all 0's, because the date is used as a dictionary key, as well as for ordering.